### PR TITLE
apps/system/utils : Modify heapinfo explanation in README.md

### PR DESCRIPTION
**Pull request by jeongchanKim**
1. apps/system/utils : Modify heapinfo explanation in README.md
    Heapinfo was expanded to show kernel, user and specific binary heap information.
    Add explanation about target for heapinfo.

2. apps/system/utils/heapinfo : Remove unnecessary description
  If CONFIG_BUILD_PROTECTED is not enabled, there is only total heap, not kernel or user.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [c8492c3986593a99fecba76129e891b3272caee4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c8492c3986593a99fecba76129e891b3272caee4] - Code Rule Check OK.

File: apps/system/utils/utils_heapinfo.c [hunk 1]
```diff
@@ -314,9 +314,7 @@ int utils_heapinfo(int argc, char **args)
 	printf("\nUsage: heapinfo [-TARGET] [-OPTION] [ARG]\n");
 	printf("Display information of heap memory\n");
 	printf("\nTargets:\n");
-#ifndef CONFIG_BUILD_PROTECTED
-	printf(" (none)         Initialize the heapinfo\n");
-#else
+#ifdef CONFIG_BUILD_PROTECTED
```
Comment (sunghan-chang) [inline apps/system/utils/utils_heapinfo.c]: When `CONFIG_BUILD_PROTECTED` is disabled, it shows like below:
```
Usage: ...
Display ...
Target:
Option:
 -a ...
```
`Targets:` is uselsess. Let's cover `Targets:` together by `CONFIG_BUILD_PROTECTED`.

File: apps/system/utils/README.md [hunk 2]
```diff
@@ -325,21 +325,26 @@ This command shows heap memory usages per thread. This has arguments which confi
 ```bash
 TASH>>heapinfo --help
 
-Usage: heapinfo [OPTIONS]
+Usage: heapinfo [-TARGET] [-OPTION] [ARG]
```
Comment (sunghan-chang) [inline apps/system/utils/README.md]: What is `ARG`?

File: apps/system/utils/README.md [hunk 3]
```diff
@@ -325,21 +325,26 @@ This command shows heap memory usages per thread. This has arguments which confi
...
-              (for -g option, CONFIG_HEAPINFO_GROUP is needed)
- -e HEAP_IDX  Show the heap[HEAP_IDX] allocation details
-              (-e option is available when CONFIG_MM_NHEAPS is greater than 1)
- -r           Show the all region information
-              (-r option is available when CONFIG_MM_REGIONS is greater than 1)
- -k OPTION    Show the kernel heap memory allocation details based on above options
+[-TARGET]
+ -k             Kernel Heap
+ -u             User Heap
+                (-u target is available when CONFIG_BUILD_PROTECTED is enabled)
```
Comment (sunghan-chang) [inline apps/system/utils/README.md]: -k also

Comment (jeongchanKim) [reply apps/system/utils/README.md]: Done. Thanks.

File: apps/system/utils/utils_heapinfo.c (hunk 1 above)
Comment (jeongchanKim) [reply apps/system/utils/utils_heapinfo.c]: Done. Thanks.

**Comment by seinfra** [comment]
Target : [992eab956ea5cdf30650b830cb5a5b29fd2bf3ac] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [992eab956ea5cdf30650b830cb5a5b29fd2bf3ac] - Code Rule Check OK.

**Pull request merged**